apps/yggdrasil/priv/repo/migrations/20160229163000_create_role_resource.exs

```diff
@@ -0,0 +1,17 @@
+defmodule Yggdrasil.Repo.Migrations.CreateRoleResource do
+  use Ecto.Migration
+
+  def change do
+    create table(:roles_resources) do
+      add :role_id, references(:roles, on_delete: :nothing)
+      add :resource_id, references(:resources, on_delete: :nothing)
+      add :permission_id, references(:permissions, on_delete: :nothing)
+
+      timestamps
+    end
+    create index(:roles_resources, [:role_id])
+    create index(:roles_resources, [:resource_id])
+    create index(:roles_resources, [:permission_id])
```

Comment:
I would drop them and keep the one for `role_id`. In general you don't want to get caught up with indexes at the beginning. The issue with indexes is you may not know which ones you need until you see what queries are being executed and what causes the query optimizer to choose a table scan.

Hey for this table, does it still have an `id` column? Seems like a composite key of role, resource, and permission would be better. Even with a large table set I don't expect these to change a lot. Plus having the persisted order based on those three columns might eliminate the need for any index even as the table grows. Which brings me back to my point about being careful with indexes. Extra ones just incur a cost at insertion time with no gain at read.